### DISCUSSION TITLE
Send all items from mini.pick to quickfix with a mapping

### DISCUSSION THREAD

**Discussion by tthkbw** [nvim-mini/mini.nvim#1028] [General]
I see how to mark items in a mini.pick window and how to send those marked items to the quickfix list. What I want to do is map both functions to one key.

For example, instead of having to use <C-a> to mark all items and then <M-CR> to send them to the quickfix list, I want to map <C-q> to do both operations. I suspect I need to create a lua function that does that and map it, but I am a lua novice.

How do I do that?

**Comment by echasnovski** [comment]
Although "mark all + choose marked" is quite a common use case, the assumption during development was that it is better to be left to atomic operations. Plus creating custom action is relatively straightforward (compared to some other customizations): either emulate pressing those two keys or use a combination of [`get_picker_matches()`](https://github.com/echasnovski/mini.nvim/blob/12d275f4c3aeddfa12a758a69aff65090613ca34/doc/mini-pick.txt#L1226) (to get matched items) and [`get_picker_opts()`](https://github.com/echasnovski/mini.nvim/blob/12d275f4c3aeddfa12a758a69aff65090613ca34/doc/mini-pick.txt#L1244) (to get picker's `choose_marked` function and execute it). Here is an example of the first one:

```lua
local choose_all = function()
  local mappings = MiniPick.get_picker_opts().mappings
  vim.api.nvim_input(mappings.mark_all .. mappings.choose_marked)
end
require('mini.pick').setup({
  mappings = {
    choose_all = { char = '<C-q>', func = choose_all },
  },
})
```

**Comment by tthkbw** [comment]
This is perfect--thanks. 

I figured it could be done easily but had no idea how to write the function. 

Now I can more easily make use of mini.pick to replace Telescope.
